drivers: touch/rtc/xt_wdt/wdt: esp32: fix conflict among device drivers

wmrsouza opened the pull request:
This PR allows to use touch_sensor, rtc_counter, xt_wdt, and wdt simultaneously by enabling
ESP_INTR_FLAG_SHARED when calling esp_intr_alloc()

uLipe commented on drivers/input/input_esp32_touch_sensor.c:
```diff
@@ -293,8 +255,8 @@ static int esp32_touch_sensor_init(const struct device *dev)
 
-	dev_data->rtc_intr_msk = ESP32_RTC_INTR_MSK;
-	esp32_rtc_isr_install(&esp32_rtc_isr, dev);
+	rtc_isr_register(esp32_touch_sensor_isr_handler, (void *)dev, ESP32_RTC_INTR_MSK, 0);
```
Is always guaranteed this register name will never change?

wmrsouza committed:
drivers: touch/rtc/wdt: esp32: fix conflict among device drivers

allows use of touch_sensor, rtc_counter, and wdt simultaneously by enabling
ESP_INTR_FLAG_SHARED when calling esp_intr_alloc()

Signed-off-by: Marcio Ribeiro <marcio.ribeiro@espressif.com>

sylvioalves commented on drivers/watchdog/wdt_esp32.c:
```diff
@@ -168,12 +168,15 @@ static int wdt_esp32_init(const struct device *dev)
+	int ret = esp_intr_alloc(
+		config->irq_source,
+		ESP_PRIO_TO_FLAGS(config->irq_priority) |
+#if IS_ENABLED(CONFIG_INPUT_ESP32_TOUCH_SENSOR) || IS_ENABLED(CONFIG_COUNTER_RTC_ESP32)
```
What if you just set `ESP_INTR_FLAG_SHARED` flag for all those peripherals regardless it is or isn't enabled? I mean, making the interrupt always shared for those 3-4 peripherals.

fabiobaltieri commented on drivers/counter/counter_esp32_rtc.c:
```diff
@@ -63,12 +63,12 @@ static int counter_esp32_init(const struct device *dev)
-				(ESP32_COUNTER_RTC_ISR_HANDLER)counter_esp32_isr,
-				(void *)dev,
-				NULL);
+	int ret =
+		esp_intr_alloc(cfg->irq_source,
+			       ESP_PRIO_TO_FLAGS(cfg->irq_priority) |
+				       ESP_INT_FLAGS_CHECK(cfg->irq_flags | ESP_INTR_FLAG_SHARED),
+			       (ESP32_COUNTER_RTC_ISR_HANDLER)counter_esp32_isr, (void *)dev, NULL);
+
```
that's some interesting indentation story going, how about adding an intermediate variable or two? Maybe just one for `ESP_PRIO_TO_FLAGS(cfg->irq_priority) | ESP_INT_FLAGS_CHECK(cfg->irq_flags | ESP_INTR_FLAG_SHARED)`

fabiobaltieri commented on drivers/input/input_esp32_touch_sensor.c:
```diff
@@ -160,17 +154,17 @@ static esp_err_t esp32_rtc_isr_install(intr_handler_t intr_handler, const void *
-			intr_handler, (void *)handler_arg, NULL);
+	err = esp_intr_alloc(
+		DT_IRQ_BY_IDX(DT_NODELABEL(touch), 0, irq),
+		ESP_PRIO_TO_FLAGS(DT_IRQ_BY_IDX(DT_NODELABEL(touch), 0, priority)) |
+			ESP_INT_FLAGS_CHECK(DT_IRQ_BY_IDX(DT_NODELABEL(touch), 0, flags) |
```
same as above

fabiobaltieri commented on drivers/watchdog/wdt_esp32.c:
```diff
@@ -168,12 +168,11 @@ static int wdt_esp32_init(const struct device *dev)
-				NULL);
+	int ret = esp_intr_alloc(
+		config->irq_source,
+		ESP_PRIO_TO_FLAGS(config->irq_priority) |
+			ESP_INT_FLAGS_CHECK(config->irq_flags | ESP_INTR_FLAG_SHARED),
```
same, arguabely the previous indentation was way better, I imagine you are using clang-format for this, sorry if you hoped that that'd save you from indentation review comments, it doesn't :-)
